keanu-project/src/main/java/io/improbable/keanu/algorithms/variational/optimizer/nongradient/NonGradientOptimizer.java

```diff
@@ -22,19 +22,44 @@
 import lombok.Builder;
 import lombok.Getter;
 
+/**
+ * This class can be used to construct a BOBYQA non-gradient optimizer.
+ * This will use a quadratic approximation of the gradient to perform optimization without derivatives.
+ * @see <a href="http://www.damtp.cam.ac.uk/user/na/NA_papers/NA2009_06.pdf">BOBYQA Optimizer</a>
+ */
 @Builder
 public class NonGradientOptimizer implements Optimizer {
-
+    /**
+     * Creates a BOBYQA {@link NonGradientOptimizer} which optimizes against the unobserved (latent) variables of the Bayesian network.
+     *
+     * @param bayesNet The Bayesian network containing the unobserved variables to be optimized against,
```

Comment:
`optimizes against the unobserved (latent) variables of the Bayesian network`

It maximizes the probability of both latent and observed variables.